src/Redux/Book/reducer.js

```diff
@@ -1,13 +1,20 @@
 const ADD_BOOKS = 'bookstore/books/ADD_BOOKS';
+const DELETE_BOOK = 'bookstore/books/DELETE_BOOK';
 
 export const addBook = (payload) => ({
   type: ADD_BOOKS,
   payload,
 });
-
+export const deleteBook = (payload) => ({
+  type: DELETE_BOOK,
+  payload,
+});
 const initialState = [];
+
 const bookReducer = (state = initialState, action) => {
   switch (action.type) {
+    case DELETE_BOOK:
+      return { ...state, deleteBook: action.filter };
```

Comment:
- [ ] DELETE_BOOK should be using the filter method. for the filter to work it should have a function/parameter that filter the array, to do so you can add `(book => book.id !== id)` as the body of the filter method, also you need to return only the one filtred book not the whole state so you can remove the specific book. kindly change your code so it works properly